osdep.c

```diff
@@ -36,7 +36,7 @@
 #include <sys/mman.h>
 #endif
 
-#ifdef CONFIG_SOLARIS
+#if defined(CONFIG_SOLARIS) && defined(_STRICT_POSIX)
```

Comment:
So the `_STRICT_POSIX` bit here is an internal flag. It's not meant to be used outside of it and was intended to be changed at any time. It can work if you need to compile on both old and new; however, if you want to say this is always synced up, then I would just get rid of the extern declaration. In this case, it should always be legal to do this both before and after without the extern. Because you can always take a `void *` and pass it as a `char *` or vice versa.

Reply:
I'll run builds with the whole block nuked to see how it fares on both before-and-after.